lib/http3/server.c

```diff
@@ -1121,7 +1121,7 @@ static int handle_input_expect_headers_process_connect(struct st_h2o_http3_serve
 
     stream->req.is_tunnel_req = 1;
     h2o_buffer_init(&stream->req_body, &h2o_socket_buffer_prototype);
-    stream->req.entity = h2o_iovec_init("", 0);
+    stream->req.entity = h2o_iovec_init(NULL, 0);
```

Comment:
Thank you for the PR.

Would it be possible to fix this on the other side, like not checking `base` when the length of the buffer is zero?

The reason is that I'm not sure if we want to make this change, because `""` and NULL have different meanings for `h2o_req_t::entity.base`. Empty string means that the request body is empty, NULL means that the request body does not exist.

While this distinction might not matter for a streaming request handler (which all CONNECT handlers have to be), there is no guarantee that CONNECT methods would be handled by a handler that does not support CONNECT - in such case, this difference might cause issues.

Reply:
Thanks! I've addressed this in 62cfde7, what do you think?